display.py

```diff
@@ -0,0 +1,308 @@
+        else:
+            syslog.syslog(
+                'Error retreiving intial %s data. It will not be shown.'
+                % plugin)
+            del(screen_info[plugin])
+
+
+# +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+while running:
+    # Look for and process keyboard events to change modes.
+    for event in pygame.event.get():
+        if event.type == pygame.KEYDOWN:
+            # On 'q' or keypad enter key, quit the program.
+            if ((event.key == pygame.K_KP_ENTER) or (event.key == pygame.K_q)):
+                running = False
+            # On 's' key, save a screen shot.
+            elif event.key == pygame.K_s:
+                my_disp.screen_cap()
+            else:
+                # On 'd' key, set mode to 'weather mode' - daily screen.
+                if event.key == pygame.K_d:
+                    mode = 'daily'
+                # On 'i' key, set mode to 'info'.
+                elif event.key == pygame.K_i:
+                    mode = 'info'
+                # on 'h' key, set mode to 'weather mode' - 'hourly'
+                elif event.key == pygame.K_h:
+                    mode = 'hourly'
+                reset_counter(mode, screen_info)
+                info_screen_time_count = 0
+                non_info_screen_time_count = 0
+
+    # Automatically switch back to weather display after info screen pause.
+    if mode not in screen_info.keys():
+        non_info_screen_time_count = 0
+        info_screen_time_count += 1
+        # Default in config.py.sample: pause for 5 minutes on info screen.
+        if info_screen_time_count > info_config.PAUSE:
+            mode = 'daily'
+            reset_counter(mode, screen_info)
+            syslog.syslog(
+                "Switching from INFO screen to DAILY screen at %s seconds"
+                % info_screen_time_count)
+    else:
+        info_screen_time_count = 0
+        non_info_screen_time_count += 1
+        # Update / Refresh the switching time after each second.
+        switch_time = time_to_switch(screen_info)
+
+        # Default in config.py.sample: flip between 2 weather screens
+        # for 15 minutes before showing info screen.
+        if non_info_screen_time_count > info_config.DELAY:
+            mode = 'info'
+            syslog.syslog("Switching to INFO screen at %s seconds"
+                          % non_info_screen_time_count)
+        elif (non_info_screen_time_count % switch_time) == 0:
+            new_screen = list(screen_info)[(
+                list(screen_info).index(mode) + 1) % len(screen_info.keys())]
+            syslog.syslog(
+                "Switching from %s screen to %s screen at %s seconds" % (
+                    mode.upper(), new_screen.upper(),
+                    non_info_screen_time_count))
+            mode = new_screen
+            screen_info[mode]['count'] += 1
+
+    if mode == 'daily' or mode == 'hourly' or mode == 'info':
```

Comment:
The previous version had recently changed to using `if MODE not in ('d', 'h'):` based on a linter suggestion... agree that that format is easier to read than what's currently here so I suggest this:

```suggestion
    if mode in ('daily', 'hourly', 'info'):
```

Reply:
This section is now removed.